packages/es-components/src/components/controls/buttons/ToggleButton.js

```diff
@@ -1,101 +1,85 @@
 /* eslint no-confusing-arrow: 0 */
 import React, { useState } from 'react';
 import PropTypes from 'prop-types';
-import styled from 'styled-components';
+import styled, { css } from 'styled-components';
 
 import Button from './Button';
+import OutlineButton from './OutlineButton';
 import { useTheme } from '../../util/useTheme';
 
+const buttonMixin = css`
+  background-color: ${props =>
+    props.isPressed ? props.variant.hoverBgColor : props.variant.bgColor};
+  color: ${props =>
+    props.isPressed ? props.variant.activeTextColor : props.variant.textColor};
+`;
+
+const StyledButton = styled(Button)`
+  .es-button--display {
+    ${buttonMixin};
+  }
+`;
+
+const StyledOutlineButton = styled(OutlineButton)`
+  ${buttonMixin};
+`;
+
 function ToggleButton(props) {
   const theme = useTheme();
   const [isPressed, setIsPressed] = useState(props.isPressed);
 
   function toggleButton(event) {
     setIsPressed(!isPressed);
-    props.handleOnClick(event);
+    props.onClick(event);
   }
 
-  const {
-    buttonClasses,
-    styleType,
-    isLinkButton,
-    size,
-    block,
-    isOutline,
-    ...buttonProps
-  } = props;
-
-  const defaultButton = {
-    bgColor: theme.colors.defaultColor,
-    textColor: theme.colors.defaultBtnText,
-    hoverBgColor: theme.colors.defaultHover,
-    hoverTextColor: theme.colors.defaultBtnText,
-    activeBgColor: theme.colors.defaultHover,
-    activeTextColor: theme.colors.defaultBtnText,
-    boxShadowColor: theme.colors.defaultHover,
-    borderColor: theme.colors.defaultColor
-  };
-  const defaultOutline = {
-    bgColor: theme.colors.white,
-    textColor: theme.colors.defaultColor,
-    hoverBgColor: theme.colors.defaultColor,
-    hoverTextColor: theme.colors.white,
-    activeBgColor: theme.colors.defaultHover,
-    activeTextColor: theme.colors.white,
-    borderColor: theme.colors.defaultColor
-  };
+  const { styleType, size, block, isOutline, ...buttonProps } = props;
 
-  let variant;
   if (isOutline) {
-    variant = theme.buttonStyles.buttonsOutline[styleType] || defaultOutline;
-  } else {
-    variant = theme.buttonStyles.buttonsNormal[styleType] || defaultButton;
+    return (
+      <StyledOutlineButton
+        {...buttonProps}
+        onClick={toggleButton}
+        styleType={styleType}
+        size={size}
+        block={block}
+        isOutline={isOutline}
+        isPressed={isPressed}
+        variant={theme.buttonStyles.outlineButton.variant[styleType]}
+      >
+        {props.children}
+      </StyledOutlineButton>
+    );
   }
 
   return (
-    <StyledToggleButton
+    <StyledButton
```

Comment:
The return here and above are identical except for the component type and variant. I think we can simplify this to a single return.

```jsx
const ToggleButtonType = isOutline ? StyledOutlineButton : StyledButton;
const styles = isOutline
  ? theme.buttonStyles.outlineButton
  : theme.buttonStyules.button;

return (
  <ToggleButtonType
    {...buttonProps}
    onClick={toggleButton}
    styleType={styleType}
    size={size}
    block={block}
    isOutline={isOutline}
    isPressed={isPressed}
    variant={styles.variant[styleType]}
  >
    {props.children}
  </ToggleButtonType>
);
```